presto-orc/src/main/java/com/facebook/presto/orc/OrcWriter.java

```diff
@@ -305,6 +307,10 @@ public OrcWriter(
                     orcEncoding.createMetadataWriter());
             columnWriters.add(columnWriter);
 
+            if (fieldType == TIMESTAMP_MICROSECONDS) {
```

Comment:
Writing different precisions works fine from writer perspective, but would fail at validate() since OrcReaderOptions support only reading in either Micro or Milli precision. I also believe we should allow writing in a single precision only. I'll look into handling this for nested types.

Reply:
We should remove the enableTimestampMicroPrecision flag and make it purely type driven. 

If writer gives type as
TIMESTAMP -> long is interpreted as millis.
TIMESTAMP_MICROS -> long is interpreted as microseconds.

Similarly on the reader it will be 
When the requested type is TIMESTAMP -> long values are of milliseconds precision.
TIMESTAMP_MICROS -> long values are of microseconds precision.